src/test/kotlin/GuessValidationTests.kt

```diff
@@ -0,0 +1,48 @@
+import org.junit.jupiter.api.Test
+import kotlin.random.Random
+import kotlin.test.assertTrue
+
+class GuessValidationTests {
+    @Test
+    fun wrongGuessLength() {
+        for (sequenceLength in MIN_SEQ_LENGTH..MAX_SEQ_LENGTH) {
+            for (colors in MIN_COLORS..MAX_COLORS) {
+                val colorsList = List(colors) { it.toString() }
+                val player = Player(sequenceLength, colorsList)
+                val guess = MutableList(sequenceLength + 1) { Random.nextInt(0, colors).toString() }
+                val guess2 = MutableList(sequenceLength - 1) { Random.nextInt(0, colors).toString() }
+                val guess3 = MutableList(sequenceLength) { Random.nextInt(0, colors).toString() }
+                assertTrue(!player.validateGuess(guess, colorsList))
+                assertTrue(!player.validateGuess(guess2, colorsList))
+                assertTrue(player.validateGuess(guess3, colorsList))
+            }
+        }
+    }
+
+    @Test
+    fun wrongGuessColors() {
+        for (sequenceLength in MIN_SEQ_LENGTH..MAX_SEQ_LENGTH) {
+            for (colors in MIN_COLORS..MAX_COLORS) {
+                val colorsList = List(colors) { it.toString() }
+                val player = Player(sequenceLength, colorsList)
+                val guess = MutableList(sequenceLength) { Random.nextInt(0, colors).toString() }
```

Comment:
używanie losowości w testach generalnie nie jest dobrym pomysłem, bo przy złym ustawieniu gwiazd i księżyca testy mogą przestać działać. A z kolei jeśli test jest napisany w taki sposób, że to nie ma znaczenia, jaki wynik się wylosuje (bo i tak go w jakiś tam sposób modyfikujecie), to równie dobrze, a nawet lepiej, jest go zderandomizować